src/fork_gc.c

```diff
@@ -761,17 +734,17 @@ static FGCError FGC_parentHandleTerms(ForkGC *gc, RedisModuleCtx *rctx) {
     return FGC_CHILD_ERROR;
   }
 
-  if (!FGC_lock(gc, rctx)) {
-    status = FGC_PARENT_ERROR;
-    goto cleanup;
+  StrongRef spec_ref = WeakRef_Promote(gc->index);
+  IndexSpec *sp = StrongRef_Get(spec_ref);
+  if (!sp) {
+    rm_free(term);
+    return FGC_SPEC_DELETED;
```

Comment:
Here you already got the `MSG_IndexInfo`, so you must do the cleanup on early return no?

Reply:
you're right